HARK/ConsumptionSaving/tests/test_IndShockConsumerType.py

```diff
@@ -922,3 +922,38 @@ def test_calc_jacobian(self):
         self.assertAlmostEqual(CJAC_Perm.T[30][29], -0.06120, places=HARK_PRECISION)
         self.assertAlmostEqual(CJAC_Perm.T[30][30], 0.05307, places=HARK_PRECISION)
         self.assertAlmostEqual(CJAC_Perm.T[30][31], 0.04674, places=HARK_PRECISION)
+
+
+# %% Compare with newer transition matrix methods
+
+
+class test_compare_trans_mats(unittest.TestCase):
+    def test_compare_will_mateo(self):
+        # No deaths for now in Mateo's method
+        # but Will's requires LivPrb < 1
+        params = deepcopy(dict_harmenberg)
+        params["LivPrb"] = [0.999]
+
+        # Create and solve agent
+        agent = IndShockConsumerType(**params)
+        agent.cycles = 0
+        agent.solve()
+        # Activate harmenberg
+        agent.neutral_measure = True
+        agent.update_income_process()
+
+        m_grid = np.linspace(0, 20, 10)
+
+        # Will's methods
+        agent.define_distribution_grid(dist_mGrid=m_grid)
+        agent.calc_transition_matrix()
+        tm_will = agent.tran_matrix
+
+        # Mateo's methods
+        agent.make_state_grid(mNrmGrid=m_grid)
+        agent.full_shock_dstns = agent.IncShkDstn
+        agent.find_transition_matrices()
+        tm_mateo = agent.trans_mats[0]
+
+        # Compare
+        self.assertTrue(np.allclose(tm_will, tm_mateo.T, atol=5e-3))
```

Comment:
Just added a test that compares the tools I am working on with @wdu9 's methods for `IndShockConsumerType`.

We are getting the same! (except I am not dealing with death/newborns yet).

Reply:
The cool feature that my tools are adding is that, to get this, I only had to define the model's state-to-state transition function.


```
    def state_to_state_trans(self, shocks_next, solution, state, PermGroFac, Rfree):


        # Consumption
        cNrm = solution.cFunc(state["mNrm"])
        # Savings
        aNrm = state["mNrm"] - cNrm


        # Shock realization
        PermGroShk = shocks_next["PermShk"] * PermGroFac
        PLvl_next = state["PLvl"] * PermGroShk
        mNrm_next = aNrm * Rfree / PermGroShk + shocks_next["TranShk"]
        
        return {
            "PLvl": PLvl_next,
            "mNrm": mNrm_next,
        }
```

This equation is now pretty expressive and could be used to simulate the model.

Reply:
Just added a way to handle newborns and the methods exactly match.